### PR TITLE
fix: ios build crashes

### PR DESCRIPTION
**Pull request by SputNikPlop**
This is a PR to get iOS builds working.
* Bumps minimum iOS version in AppFrameworkInfo.plist
* Adds Google Mobile Ads to Info.plist https://developers.google.com/admob/flutter/quick-start#ios
* Firebase initialization https://firebase.flutter.dev/docs/overview/#initialization
* Mobile ads initialization https://developers.google.com/ad-manager/mobile-ads-sdk/flutter/quick-start#ios

The mobile ads have to initialized to prevent crashes like it was in https://stackoverflow.com/questions/61580053/why-does-my-flutter-app-crash-at-startup-on-ios

The app builds iPhone 14 iOS 16.2
Simulator Screen Shot - iPhone 14 - 2023-01-29 at 10 28 08

There is still a few things to solve (don't have the time right now)
* Figure out black spaces on top and bottom
* Mobile Ads SDK https://developers.google.com/admob/ios/ios14#skadnetwork

I know you don't have access to a mac to test but I hope these contributions help the project somewhat. Thank you as well for the contributor credit that you added to the application as well.

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
 import 'package:jais/views/initialization_view.dart';
 import 'package:provider/provider.dart';
 
-void main() {
+void main() async {
   WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp(
+    options: DefaultFirebaseOptions.currentPlatform,
+  );
+  await MobileAds.instance.initialize();
```
Comment (Ziedelth) [inline lib/main.dart:22]: Normally, Firebase is initialized in AppController.
For MobileAds, it initializes itself when a video is loaded.

**Review by Ziedelth — CHANGES_REQUESTED** [review]

**Comment by Ziedelth** [comment]
For the black spaces at the top and bottom of the application, I think it comes from the SafeArea in HomeView.
Looking deeper, I don't have this concern with Android.

File: lib/main.dart (hunk 1 above)
Comment (SputNikPlop) [reply lib/main.dart]: Good catch, my bad.

**Comment by SputNikPlop** [comment]
> For the black spaces at the top and bottom of the application, I think it comes from the SafeArea in HomeView. Looking deeper, I don't have this concern with Android.
> 
> EDIT : Apparently, other people have had the same problem, and it could be the Scaffold.resizeToAvoidBottomInset (https://stackoverflow.com/questions/64458564/flutter-app-leaves-blank-space-at-the-top-of-the-screen-when-removing-system-ove). To be tested further.

This helps a lot hopefully I may get to it in the near future.

File: lib/main.dart [hunk 2]
```diff
@@ -11,9 +11,8 @@ import 'package:jais/views/animes/anime_search_view.dart';
 import 'package:jais/views/initialization_view.dart';
 import 'package:provider/provider.dart';
 
-void main() {
+void main() async {
```
Comment (Ziedelth) [inline lib/main.dart:14]: You can remove the "async", as there is no more "await" in the

File: lib/controllers/app_controller.dart [hunk 3]
```diff
@@ -6,6 +6,13 @@ import 'package:jais/controllers/datas/data_collection_controller.dart';
 import 'package:jais/controllers/url_controller.dart';
 import 'package:jais/firebase_options.dart';
 
+void main() async {
+  WidgetsFlutterBinding.ensureInitialized();
```
Comment (Ziedelth) [inline lib/controllers/app_controller.dart:10]: WidgetsFlutterBinding is already initialized in the main method of main.dart.
It is not, therefore, necessary to reset it.

File: lib/controllers/app_controller.dart [hunk 4]
```diff
@@ -6,6 +6,13 @@ import 'package:jais/controllers/datas/data_collection_controller.dart';
...
+  await Firebase.initializeApp(
+    options: DefaultFirebaseOptions.currentPlatform,
+  );
```
Comment (Ziedelth) [inline lib/controllers/app_controller.dart:13]: As said before, Firebase is initialized in the initialize method of the controller.

**Review by Ziedelth — CHANGES_REQUESTED** [review]

**Review by Ziedelth — APPROVED** [review]

**Pull request merged**